### PR TITLE
Installation docs

### PR DESCRIPTION
**Pull request by Yoshanuikabundi**
This PR adds a page of documentation describing how to install OpenFF software via MambaForge to the ecosystem-level documentation. Much of this is duplicated in all our project docs, but my hope is with this we can reduce most of those docs to "Run this conda install command. If you don't have conda, see \[](bla)".

A lot of this is adapted from the [toolkit install guide](https://docs.openforcefield.org/projects/toolkit/en/stable/installation.html), but I've updated it with advice that I think makes a more stable Conda installation. Parts of it contradict the Toolkit guide, so we should update that if we like these instructions.

I think I've managed to get comprehensive instructions that should suit 80%+ of complete newcomers into the first rendered page of the document (as in, you don't need to scroll on a desktop to read it). The rest of the page explains why we do things that way, covers some edge cases like Windows, Apple Silicon, and existing Conda installations, and finally describes environment management in quite a lot of detail. I expect most users will not read this - hopefully they get what they need from the quick install guide. But it gives us a place to send people when they ask about some of the trickier parts of Conda!

Questions I have:
1. Does the quick install guide at the top do the job? Are there ways to further simplify it that won't leave users unsure about things?
2. I struggled a lot with the whole question of whether to tell users to say yes to initialization. Saying "no" is what we've historically advised in the Toolkit docs, but it makes using everything a huge pain as you have to enter like 3 commands just to get started with a shell session. Saying "yes" and then turning off `auto_activate_base` is what I do, and it's extremely convenient for ongoing use, but it does almost double the length of the quick install guide. An alternative approach I thought about was to ignore the question altogether in the quick install guide, and leave it up to the user (with the additional advice later in the page). Is there a better approach?
3. The quick install guide only covers how to install a package in a general sense. You need an additional piece of information to apply this to OpenFF packages: The package name. At the moment this is provided via a link to the projects list. Is there a better way?

@xyzmjf

### CODE REVIEW

File: source/install.md
Comment (mattwthompson) [inline source/install.md:110]: Suggest adding something like

> Virtual environments are designed to be rapidly deployable, which also means they can quickly and safely be removed and re-created in a matter of seconds in case something goes wrong.

Personal preference to emphasize some smithed version of this, I regularly find that beginners waste days of their lives trying to fix weird states when blowing everything away and starting fresh takes a few minutes.

Comment (mattwthompson) [reply source/install.md:110]: Oh, maybe the following paragraph is a better location ...

File: source/install.md
Comment (mattwthompson) [inline source/install.md:160]: > tinker with ...

If this implies "don't muck around in `/site-packages/`" we should have stronger language against doing this. If you're referring to editable installs while switching branches between projects ... ugh, that's annoying and I keep doing it.

File: source/install.md
Comment (mattwthompson) [inline source/install.md:168]:
```suggestion
But that can go wrong if you're mixing channels and might be slow if the environment is old. Recreating the environment often works better:
```

File: source/install.md
Comment (mattwthompson) [inline source/install.md:189]: > The catch is that the environment may solve and install but not work!

Is this true? I'm not sure I've ever had this happen to me. Usually my issues with mixed channels relate to the environment not solving at all.

File: source/install.md
Comment (mattwthompson) [inline source/install.md:209]:
```suggestion
OpenFF does not support or test on Windows natively. All of our software is pure Python code and may work anyway as long as you can provide our dependencies, but many of these dependencies are not available on Windows and may not be in the foreseeable future.
```

Comment (mattwthompson) [reply source/install.md:209]: Personally prefer emphasizing that we CURRENTLY cannot do anything on Windows because of our upstreams, even though this is a cop-out it shifts the blame away from us and puts a bit of distance between us and the appearance that Windows is not supported because we didn't bother trying.

File: source/install.md
Comment (mattwthompson) [inline source/install.md:211]:
```suggestion
Instead, we recommend using the [Windows Subsystem for Linux] to run OpenFF software on Windows. WSL runs a Linux kernel within your Windows system so you can run ordinary software as if you had a Linux system. If your hardware supports it, we suggest using WSL2 for a smoother experience; this is the default for new WSL installations on supported hardware. WSL2 requires hardware virtualization support, which is available on most modern CPUs but may require activation in the BIOS/UEFI. 
```

**Comment by mattwthompson** [comment]
I'm undecided on whether or not I'm a +1 on this change. I'm not a -1 but I'm somewhere in the grey. I'll happily defer to @j-wags on everything, especially seeing as that's his job.

* I see the value to new/new-ish users of holding their hands through a conda installation
  * I think it's fair for us to offer some guidance on how to get set up
  * I hate being responsible for every single thing that goes wrong in user's conda installations, usually stemming from corner cases that can't be covered in documentation like this.
* I think this is a good place to store some of this information
  *  it's certainly a better place to go into this type of detail than i.e. a subsection in the toolkit's installation docs.
* I love to see project-specific installation guides as short as possible. (I'm much more likely to use somebody's tool if I can find the installation instructions, read them in a couple of seconds, and actually do the installation in a one-liner.) We're already [in pretty good shape there](https://docs.openforcefield.org/projects/interchange/en/stable/installation.html) but this should help trim things down a little bit.
* I worry about this drifting out of date, partially because so many (currently correct! and currently great!) detailed recommendations are included here. The ecosystem is better and more stable than it was 5 years ago, but if Quansight engineers come out next month with something that's drastically better but requires changes to our setup, I would not be surprised.
* We should to be diligent about avoiding duplicated information between this and project-specific installation documentation.
  *  I'm not sure how best to handle this; if I was a scientist that just wanted to see if Sage covers my ligand, there's a limit to the number of documentation pages I'm going to click through to get that done. (I'm also lazy ... maybe it's not a coincidence that I'm no longer a scientist?).
  * It would be very not-good for details to diverge, i.e. if we introduce Windows support and testing but two similar pages give different status of that.
* Are there cases in which recommendations might differ between here and individual projects? I can't think of any examples in the projects I'm currently focused on, it's not like I'm going to recommend against `mamba` for Interchange. If such cases exist, I guess the project-specific recommendations take precedence, but I'm not sure. Maybe case-by-case/cross that bridge when we get there.

> Does the quick install guide at the top do the job? Are there ways to further simplify it that won't leave users unsure about things?

I don't think there's much to improve here. The second sentence in the only one I'd consider cutting since it's not necessarily something users need to know to get started. Bullet points 2a and 2c repeat the installers' defaults but are worth including because users don't always know if they should trust defaults.

> I struggled a lot with the whole question of whether to tell users to say yes to initialization.

I've always said "yes" and it's always worked fine for me, so I'm not sympathetic to the arguments for saying "no." My experience may not be typical. Your alternative suggestion seems fine to me; if you absolutely need a tiebreak then I lean towards preferring how it is, favoring explicitness is beginner-facing instructions even if repeating what's described in the installer.

> The quick install guide only covers how to install a package in a general sense ...

I think it's fine as-is. If our project were independent of each other, I think it would be useful to have a section that lays this out. But, as it is, just installing the toolkit should cover a majority of use cases. I think the complication arises when somebody wants to install BespokeFit and I sure don't like the idea of squeezing that complexity into subsection 3b. Currently, your one-liner serves everybody that needs to use the toolkit but doesn't need special bits from Evaluator, QCSubmit, or Recharge - in other words, the vast majority of non-BespokeFit users.

File: source/install.md
Comment (Yoshanuikabundi) [reply source/install.md:73]: Good call, I'll figure out a way to do that. I figured if they're happy to share curl commands with name interpolation of these URLs they can't be too likely to change!

File: source/install.md
Comment (Yoshanuikabundi) [reply source/install.md:85]: I've been burned a couple of times by systems failing to update because a package manager was expecting the system Python and got the base Conda environment instead. I know most package managers are more resilient than that but I really don't want to create weird problems for our users.

File: source/install.md
Comment (Yoshanuikabundi) [reply source/install.md:160]: Yeah referring to the former, I'll strengthen and clarify.

File: source/install.md
Comment (Yoshanuikabundi) [reply source/install.md:189]: I'm sure I had a problem last week where a BespokeFit+Psi4 environment solved with flexible priority, but then I got missing symbol errors when trying to run it. And I believe this is one of the motivations behind strict. But it's working today so maybe I'm misremembering.

File: source/install.md
Comment (Yoshanuikabundi) [reply source/install.md]: Good call - I love how your phrasing warns that users are gonna have a bad time without feeling like we're saying they're breaking an arbitrary rule.

**Comment by Yoshanuikabundi** [comment]
> I see the value to new/new-ish users of holding their hands through a conda installation 

Yeah Mark (@xyzmjf) and I agree that this is really important for helping new users get started, and it's not the last getting started guide we want to add to the ecosystem docs. 

> I hate being responsible for every single thing that goes wrong in user's conda installations, usually stemming from corner cases that can't be covered in documentation like this.

Yeah it sucks. I think the Toolkit already sort of takes on that responsibility though, and this is a better, more findable place for it.

> I'm much more likely to use somebody's tool if I can find the installation instructions, read them in a couple of seconds, and actually do the installation in a one-liner.

Yep!!!! There's a tension between instructions that a beginner can get started with at all, and that an expert can parse quickly enough that it doesn't turn them off. Hopefully a formulation like "To install, run conda install bla. If that doesn't make sense, see here" splits the difference.

> I worry about this drifting out of date, partially because so many (currently correct! and currently great!) detailed recommendations are included here.

I think if something new and amazing happens, these instructions will at least mean that our users have robust set ups that are similar to our own - which should make migration easier to manage and support. My gut says that this guide isn't going to fall apart - it might just drift (or lurch!) away from best practice. And centralising this information and linking to it from project docs will make it much easier to update!

> We should to be diligent about avoiding duplicated information between this and project-specific installation documentation. 

Agreed - once this is merged I'll open PRs in all our projects to simplify their install guides and link here.

> Are there cases in which recommendations might differ between here and individual projects?

Yeah, I don't think such cases exist yet, which to me suggests some level of confidence that they won't suddenly appear. We have a LOT of packages, and by now I'd like to think I've more or less figured out how the installation guides go - every time I write a new one I have an internal fight about how much of this information to include, but not about whether this information is relevant or correct. So I'm pretty sure this is close to the maximum installation information shared by all theoretical Openff packages (and most conda forge packages at large).

> I don't think there's much to improve here 

> if you absolutely need a tiebreak then I lean towards preferring how it is 

> I think it's fine as-is. 

:napoleon_dynamite_fist_pump:

File: source/install.md
Comment (Yoshanuikabundi) [reply source/install.md:110]: Done in 86c1c84

File: source/install.md
Comment (Yoshanuikabundi) [reply source/install.md]: Done in 31bd01f

**Comment by j-wags** [comment]
Let's chat about this at our check in tomorrow. Overall I think it's in great shape but I want to think more about a few things:

* generally I DON'T recommend initting so I want to think about this rec
* some hesitation over calling them "virtual environments" since that will invite confusion with actual python virtual environments
* I think we could really drive the point home that this doesn't require sudo/root access. New users are really confused by this. It's not clear to me that the `local to your user` link clarifies this
* It may be good to clarify that mamba needs to go into the base env
* I think the quick start guide is supposed to end right before the header `Installing Mamba`, but I don't see visual cues that it does. So it may be good to have some sort of stronger break - even the sentence "that's it, you're done with the quick start guide, everything below is on handling more advanced installs/situations"

Anyway, I'm inclined toward approving since this seems like quite an improvement, I'd just like to think about it a bit more and talk it out. The init thing is my only strongish opinion.

**Comment by Yoshanuikabundi** [comment]
> some hesitation over calling them "virtual environments" since that will invite confusion with actual python virtual environments

[Conda itself](https://docs.conda.io/projects/conda/en/latest/user-guide/concepts/environments.html#virtual-environments) uses this language to describe its environments, if that helps. They're pretty interchangeable; for instance, you can provide a Conda environment to PyRight as though it were a `venv` and it works fine. 

> I think we could really drive the point home that this doesn't require sudo/root access. New users are really confused by this. It's not clear to me that the local to your user link clarifies this

Mmm good idea.

> It may be good to clarify that mamba needs to go into the base env

Will do

> I think the quick start guide is supposed to end right before the header Installing Mamba, but I don't see visual cues that it does

Will try some things out

File: source/install.md
Comment (Yoshanuikabundi) [reply source/install.md]: Added a CI workflow that will tell us if any links break in the built documentation!

**Review by j-wags — APPROVED** [review]
@Yoshanuikabundi and I chatted during our check in today about the "to init or not to init" question, and we resolved that we'll put the "warning: if you already have conda or mamba, skip steps 1+2" warning ABOVE the install steps. This should avoid the case where some industry user has an existing conda installation that they don't understand and we "break" it by telling them to init 

(in an ideal world, users would have a better handle on their environments, but especially at the companies that pay us folks are very busy and wouldn't look favorably on a quick start guide that breaks their other project software)

Also (not blocking) I like the addition of the horizontal line, but I think there still could be even more of a separation after the quick start guide - especially something containing the phrase "everything else here is advanced installation instructions or details"

File: source/install.md
Comment (xyzmjf) [reply source/install.md]: Hi There. FYI I had a day off yesterday, so only just starting to look at this. I will send my comments to Josh by email.

**Comment by Yoshanuikabundi** [comment]
New build for `openff-toolkit-examples` 0.12.1 is out and can run the Toolkit Showcase via the instructions in this PR. Merging! Thanks for everyone's suggestions!

**Pull request merged**
